### PR TITLE
Bring in Group.Name API from future branch, to be supported for net463 

### PR DESCRIPTION
**Pull request by Priya91**
Going to be adding new APIs that shipped in 4.1.0, and the one from future to desktop 463, hence merging to master. Also bumped up the contract version to 4.2.0.

cc @joshfree @ericstj

### CODE REVIEW

File: src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.1.1.0</AssemblyVersion>
+    <AssemblyVersion>4.2.0.0</AssemblyVersion>
```
Comment (weshaggard) [inline src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.csproj]: This may break the TFS build when it mirrors.

**Comment by bartonjs** [comment]
@dotnet-bot Test this please (execution seems to have been abandoned)

File: src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.csproj (hunk 1 above)
Comment (Priya91) [inline src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.csproj]: Did a TFS build with just the package changes to 4.2.0.0, the build passed..

Comment (weshaggard) [inline src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.csproj]: Thanks for verifying.

**Comment by Priya91** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

**Comment by Priya91** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Comment by Priya91** [comment]
System.Net.Http,Functional.Tests failing breaking PRs, #9558

**Comment by Priya91** [comment]
@dotnet-bot test ci please

**Comment by Priya91** [comment]
@dotnet-bot test this please

**Pull request merged**